### ISSUE TITLE
Model()[1].Paremeters() does not work in fnn 

### ISSUE THREAD

**Issue by RaffaeleMarino** [mlpack/mlpack#1906]
Hi, 
I am new in mlpack and I am trying to use them for research.
I need to print the weights of a layer of a neural networks. On the documentation at the bottom of the page https://www.mlpack.org/doc/stable/doxygen/anntutorial.html it has been written :

To access the weights from the neural network layers, you can call the following function on any initialized network:

model.Parameters();

which will return the complete model parameters as an armadillo matrix object; however often it is useful to not only have the parameters for the complete network, but the parameters of a specific layer. Another method, Model(), makes this easily possible:

model.Model()[1].Parameters();

In the example above, we get the weights of the second layer.

However, if I call this function for my FNN I have a compile error:

main.cpp: In function ‘int main(int, char**)’:
main.cpp:208:17: error: ‘__gnu_cxx::__alloc_traits<std::allocator<mlpack::ann::FFN<mlpack::ann::MeanSquaredError<>, mlpack::ann::RandomInitialization> >, mlpack::ann::FFN<mlpack::ann::MeanSquaredError<>, mlpack::ann::RandomInitialization> >::value_type’ {aka ‘class mlpack::ann::FFN<mlpack::ann::MeanSquaredError<>, mlpack::ann::RandomInitialization>’} has no member named ‘Model’
   cout<<MYNN[j].Model()[0].Parameters()

My FNN is a stack of sub-FNN
for (size_t i = 1; i < t_max-1; i++) {
  FFN<MeanSquaredError<>, RandomInitialization> model;
    model.Add<Linear<> >(dim, dim + 10);
    model.Add<BatchNorm<> >(dim + 10);
    model.Add<ReLULayer<> >();
    model.Add<Linear<> >(dim + 10, dim);
    model.Add<BatchNorm<> >(dim);
    model.Add<ReLULayer<> >();
    MYNN.push_back(model);
  }

Many thanks 

  

#### Your environment

 * version of mlpack:
 * operating system:
 * compiler:
 * version of dependencies (Boost/Armadillo):
 * any other environment information you think is relevant:

**Comment by jeffin143** [comment]
@sreenikSS , check this out, I guess @RaffaeleMarino , undergoes with the same issue over which we had the discussion. I am pretty sure now that it isn't there.

**Comment by sreenikSS** [comment]
@jeffin143 you are right. This needs to be promptly addressed.

**Comment by zoq** [comment]
@RaffaeleMarino Thanks for the report, looks like the network isn't accessible via the mentioned Model() function.

For a quick solution you could add:

```c++
const std::vector<LayerTypes<CustomLayers...> >& Model() const { return network; }
```

to the ``FFN`` class or if you don't like to rebuild mlpack:

```c++
#define private public

model.network.[1].Parameters()
```

could work as well; note this is not a safe solution.

@jeffin143 @sreenikSS do you have something else in mind?

**Comment by RaffaeleMarino** [comment]
@zoq @jeffin143 @sreenikSS  Thank you very much for your advice and your time. I will see if I can rebuild mlpack, in the meantime I will check if the solution

#define private public

model.network.[1].Parameters()

works. 

Raffaele
